test/test_torch.py

```diff
@@ -4323,6 +4323,9 @@ def test_assertEqual(self):
         self.assertEqual(x, xv)
         self.assertEqual(xv, x)
 
+        self.assertRaisesRegex(AssertionError, "don't combine", 
+                               lambda: self.assertEqual(x, xv, 1.0, ""))
```

Comment:
@ezyang this is the test for your comment about verifying "they'll get a weird error message here, right?"